#########################################################################
# ⭐⭐⭐  Set parameters carefully for satisfied performance!   ⭐⭐⭐ #
#########################################################################

mass                : 2.2 # kg 
gravity             : 9.805 
ctrl_freq_max       : 100.0
max_manual_vel      : 4.0


msg_timeout:
    state           : 0.5
    rc              : 0.5
    imu             : 0.5
    battery         : 0.5
    odom            : 0.5
    cmd             : 0.5


rc_reverse:
    roll            : false
    pitch           : false
    yaw             : false
    throttle        : false


thrust_model: # The model that maps thrust signal u(0~1) to real thrust force F(Unit:N): F=K1*Voltage^K2*(K3*u^2+(1-K3)*u). 
    accurate_thrust_model: false  # This can always enabled if don't require accurate control performance :-)
    # accurate thrust mapping parameters
    K1: 0.7583 # Needs precise calibration!
    K2: 1.6942 # Needs precise calibration!
    K3: 0.6786 # Needs precise calibration! K3 equals THR_MDL_FAC in https://docs.px4.io/master/en/config_mc/pid_tuning_guide_multicopter.html.
    # approximate thrust mapping parameters
    hover_percentage: 0.72  # Thrust percentage in Stabilize/Arco mode # *


gain: 
    # Cascade PID controller. Recommend to read the code.
    Kp0: 1.5
    Kp1: 1.5 
    Kp2: 1.5
    Kv0: 1.5
    Kv1: 1.5
    Kv2: 1.5
